youtube_dl/extractor/roosterteeth.py

```diff
@@ -54,86 +54,110 @@ def _login(self):
 
-        webpage = self._download_webpage(url, display_id)
-
-        episode = strip_or_none(unescapeHTML(self._search_regex(
-            (r'videoTitle\s*=\s*(["\'])(?P<title>(?:(?!\1).)+)\1',
-             r'<title>(?P<title>[^<]+)</title>'), webpage, 'title',
-            default=None, group='title')))
-
-        title = strip_or_none(self._og_search_title(
-            webpage, default=None)) or episode
-
-        m3u8_url = self._search_regex(
-            r'file\s*:\s*(["\'])(?P<url>http.+?\.m3u8.*?)\1',
-            webpage, 'm3u8 url', default=None, group='url')
-
+        headers = {}
+        if self._ACCESS_TOKEN:
+            headers['Authorization'] = 'Bearer {}'.format(self._ACCESS_TOKEN)
+
+        api_response = self._call_api(
+            display_id,
+            note='Downloading video information (1/2)',
+            errnote='Unable to download video information (1/2)',
+            headers=headers,
+        )
+
+        if len(api_response.get('data', [])) == 0:
+            raise ExtractorError('Unable to download video information')
+        data = api_response.get('data')[0]
+
+        attributes = data.get('attributes', {})
+        episode = attributes.get('display_title')
+        title = attributes.get('title')
+        description = attributes.get('caption')
+        series = attributes.get('show_title')
+
+        images = data.get('included', {}).get('images')
+        if images and len(images) > 0:
+            images = images[0]
+        thumbnail = images.get('attributes', {}).get('thumb')
+
+        video_response = self._call_api(
+            display_id,
+            path='/videos',
+            note='Downloading video information (2/2)',
+            errnote='Unable to download video information (2/2)',
+            headers=headers,
+        )
+
+        if video_response.get('access') is not None:
+            now = time.time()
+            sponsor_golive = unified_timestamp(attributes.get('sponsor_golive_at'))
+            member_golive = unified_timestamp(attributes.get('member_golive_at'))
+            public_golive = unified_timestamp(attributes.get('public_golive_at'))
+
+            if attributes.get('is_sponsors_only', False):
+                if now < sponsor_golive:
+                    self._golive_error(display_id, 'FIRST members')
+                else:
+                    self.raise_login_required('{0} is only available for FIRST members'.format(display_id))
+            else:
+                if now < member_golive:
+                    self._golive_error(display_id, 'site members')
+                elif now < public_golive:
+                    self._golive_error(display_id, 'the public')
+                else:
+                    raise ExtractorError('Video is not available')
+
+        if len(video_response.get('data', [])) == 0:
+            raise ExtractorError('Unable to download video information')
+        video_attributes = video_response.get('data')[0].get('attributes')
```

Comment:
Same.

Reply:
I'm not sure what 'Same' is meant to be referencing, but I removed the check and error, and changed video_attributes declaration to use try_get. If this is not what you meant for me to change, I can revert it.